lib/astutils.cpp

```diff
@@ -1369,6 +1369,10 @@ bool isVariableChangedByFunctionCall(const Token *tok, int indirect, const Setti
         parenTok = parenTok->link()->next();
     const bool possiblyPassedByReference = (parenTok->next() == tok1 || Token::Match(tok1->previous(), ", %name% [,)}]"));
 
+    if (tok->isKeyword() && !Token::Match(tok, "reinterpret_cast|static_cast|dynamic_cast|const_cast")) {
```

Comment:
would it be possible to use isCast here?

Reply:
That actually doesn't work, unfortunately. Only the paren `(` is set for cast not the keyword.

Reply:
hmm.. how about: `tok->astParent() && isCPPCast(tok->astParent())`?

Reply:
I think the parent is `<`, let me check.